File: ws/arboleda/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Feline.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Abner Arboleda, Logic Legion, DCCO-ESPE
 */
public abstract class Feline extends Mammal{

    public Feline(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }

}

File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Feline.js
const Mammal = require('./Mammal');

class Feline extends Mammal {

    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
        if (new.target === Feline) {
            throw new Error('Cannot instantiate abstract class Feline');
        }
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
}

module.exports = Feline;